hadoop-common-project/hadoop-common/src/site/markdown/filesystem/filesystem.md

```diff
@@ -599,6 +599,25 @@ on the filesystem.
     if len(FS, P) > 0:  getFileStatus(P).getBlockSize() > 0
     result == getFileStatus(P).getBlockSize()
 
+1. The outcome of this operation MUST be identical to the value of
+   `getFileStatus(P).getBlockSize()`.
+2. By inference, it MUST be > 0 for any file of length > 0.
+###  `Path getEnclosingRoot(Path p)`
+
+This method is used to find a root directory for a path given. This is useful for creating 
+staging and temp directories in the same root directory. There are constraints around how 
+renames are allowed to atomically occur (ex. across hdfs volumes or across encryption zones).
+
+#### Preconditions
+
+if no root path is found:  raise IOException
+No root path would be found only if there is no mount point for the given path
+
+
+#### Postconditions
+
+The path return will not be null
```

Comment:
better

a path representing the "root" directory for a mounted fs

Reply:
added more pre/post, should be more clear now and included some of what you suggested above